File: Fines2Model3/DAO/Toma.cs
using SqlOrganize.CollectionUtils;

namespace SqlOrganize.Sql.Fines2Model3
{
    public class TomaDAO
    {
        public static IEnumerable<Toma> TomasAprobadas__By_IdCalendario(object idCalendario)
        {
            string sql = @"
                SELECT DISTINCT toma.id 
                INNER JOIN curso ON toma.curso = curso.id
                INNER JOIN comision ON curso.comision = comision.id
                INNER JOIN calendario ON comision.calendario = calendario.id
                WHERE calendario.id = @IdCalendario
                AND (estado = 'Aprobada' OR estado = 'Pendiente')
                AND estado_contralor != 'Modificar'
";

            return Context.db.CacheSql().QueryIds<Toma>(sql, new { IdCalendario = idCalendario });
        }

        public static IEnumerable<Toma> TomasNoAprobadas__By_IdCalendario(object idCalendario)
        {
            string sql = @"
                SELECT DISTINCT toma.id 
                INNER JOIN curso ON toma.curso = curso.id
                INNER JOIN comision ON curso.comision = comision.id
                INNER JOIN calendario ON comision.calendario = calendario.id
                WHERE calendario.id = @IdCalendario
                AND estado != 'Aprobada'
                AND estado_contralor = 'Modificar'
";

            return Context.db.CacheSql().QueryIds<Toma>(sql, new { IdCalendario = idCalendario });
        }

        public static IEnumerable<Toma> TomasAprobadas_By_IdCurso(params object[] idsCursos)
        {
            string sql = @"
                SELECT DISTINCT toma.id 
                INNER JOIN curso ON toma.curso = curso.id
                INNER JOIN comision ON curso.comision = comision.id
                INNER JOIN calendario ON comision.calendario = calendario.id
                WHERE curso IN ( @IdsCursos ) 
                AND estado = 'Aprobada'
                AND estado_contralor != 'Modificar'
";

            return Context.db.CacheSql().QueryIds<Toma>(sql, new { IdsCursos = idsCursos });
        }

        public static IEnumerable<Toma> TomasAprobadas__By_IdsComisiones(params object[] idsComisiones)
        {
            string sql = @"
                SELECT DISTINCT toma.id 
                INNER JOIN curso ON toma.curso = curso.id
                WHERE curso.comision IN ( @IdsComisiones ) 
                AND estado = 'Aprobada'
                AND estado_contralor != 'Modificar'
";

            return Context.db.CacheSql().QueryIds<Toma>(sql, new { IdsComisiones = idsComisiones });
        }

        public static IEnumerable<Toma> TomasPasar__By_IdCalendario(object idCalendario)
        {
            string sql = @"
                SELECT DISTINCT toma.id 
                INNER JOIN curso ON toma.curso = curso.id
                INNER JOIN comision ON curso.comision = comision.id
                WHERE comision.calendario = @IdCalendario 
                AND estado_contralor = 'Pasar'
                AND ($estado = 'Aprobada' OR $estado = 'Renuncia' OR $estado = 'Baja')
";
            return Context.db.CacheSql().QueryIds<Toma>(sql, new { IdCalendario = idCalendario });

        }


        public static IEnumerable<Toma> TomasPasar__By_IdCalendario__Without_PlanillaDocente(object idCalendario)
        {
            string sql = @"
                SELECT DISTINCT toma.id 
                INNER JOIN curso ON toma.curso = curso.id
                INNER JOIN comision ON curso.comision = comision.id
                INNER JOIN docente ON toma.docente = docente.id
                INNER JOIN persona ON docente.persona = persona.id
                LEFT JOIN asignacion_planilla_docente ON asignacion_planilla_docente.toma = toma.id
                WHERE comision.calendario = @IdCalendario 
                AND estado_contralor = 'Pasar'
                AND ($estado = 'Aprobada' OR $estado = 'Renuncia' OR $estado = 'Baja')
                AND asignacion_planilla_docente.id IS NULL
                ORDER BY persona.numero_documento ASC
            ";  
            return Context.db.CacheSql().QueryIds<Toma>(sql, new { IdCalendario = idCalendario });

        }
    }
}

File: Data/Toma_.php
<?php

namespace Fines2;

use \Fines2\Toma;

use SqlOrganize\Sql\DbMy;

class Toma_ extends Toma
{
public static function TomasByCalendario($calendario): array {
        $db = DbMy::getInstance();

        $dataProvider = $db->CreateDataProvider();

        $sql = "
            SELECT DISTINCT id 
            FROM toma
            INNER JOIN curso ON (toma.curso = curso.id)
            INNER JOIN comision ON (curso.comision = comision.id)
            INNER JOIN calendario ON (comision.calendario = calendario.id)
            WHERE calendario = :calendario
        ";

        return $dataProvider->fetchEntitiesBySqlId("toma", $sql, ["calendario" => $calendario]);
    }


    public static function TomasActivasByCalendario($calendario): array {
        $db = DbMy::getInstance();

        $dataProvider = $db->CreateDataProvider();

        $sql = "
            SELECT DISTINCT toma.id 
            FROM toma
            INNER JOIN curso ON (toma.curso = curso.id)
            INNER JOIN comision ON (curso.comision = comision.id)
            INNER JOIN calendario ON (comision.calendario = calendario.id)
            WHERE calendario = :calendario
        ";

        return $dataProvider->fetchEntitiesBySqlId("toma", $sql, ["calendario" => $calendario]);
    }

    public static function TomasContralorByCalendario($calendario): array {
        $db = DbMy::getInstance();

        $dataProvider = $db->CreateDataProvider();

        $sql = "
            SELECT DISTINCT toma.id 
            FROM toma
            INNER JOIN curso ON (toma.curso = curso.id)
            INNER JOIN comision ON (curso.comision = comision.id)
            INNER JOIN persona ON (toma.docente = persona.id)
            WHERE comision.calendario = :calendario
            AND (toma.estado = 'Aprobada') 
            AND toma.estado_contralor = 'Pasar'
            AND (toma.planilla_docente IS NULL OR toma.reclamo = true)
            ORDER BY persona.numero_documento ASC;
        ";

        return $dataProvider->fetchEntitiesBySqlId("toma", $sql, ["calendario" => $calendario]);
    }
}

